pysys/baserunner.py

```diff
@@ -160,6 +164,17 @@ def __init__(self, record, purge, cycle, mode, threads, outsubdir, descriptors,
 		else:
 			self.writers.append(ConsoleSummaryResultsWriter())
 		
+		self.__collectTestOutput = []
+		for c in self.project.collectTestOutput:
+			c = dict(c)
+			assert c['outputDir'], 'collect-test-output outputDir cannot be empty'
+			c['outputDir'] = os.path.join(self.project.root, c['outputDir']\
+				.replace('@OUTDIR@', os.path.basename(self.outsubdir)) \
```

Comment:
Yes indeed. I gave this a lot of thought as I prefer consistency, but basically everything else (including substitution of hostname, timestamps, and anything user-defined) is dealt with in a uniform, simple and user-extensible way using the ${..} property expansion mechanism. But can't use that for outdir since it's known a long time after the proeprty expansion process (part of project file parsing) has concluded, only after the runner constructor has completed. So it does need to be done differently (sadly!). Note that this @OUTDIR@ syntax is already used for the performance reporter filename so we're just using the same mechanism here for the coverage dir.

Reply:
OK